pkgs/applications/misc/rtfm/default.nix

```diff
@@ -0,0 +1,66 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, fetchurl
+, callPackage
+, crystal_1_8
+, libadwaita
+, wrapGAppsHook4
+, gobject-introspection
+, desktopToDarwinBundle
+, webkitgtk_6_0
+, sqlite
+, gi-crystal
+}:
+let
+  gtk-docs = callPackage ./gtk-docs.nix { };
+in
+crystal_1_8.buildCrystalPackage rec {
+  pname = "rtfm";
+  version = "0.2.1";
+
+  src = fetchFromGitHub {
+    owner = "hugopl";
+    repo = "rtfm";
+    rev = "v${version}";
+    name = "rtfm";
+    hash = "sha256-ZhKwnfoU3LgCniJn3Ino1MlHfmPQCyz9LLt+7ci9iBk=";
+  };
+
+  crystal_doc_version = "1.9.0";
+
+  crystal-docs = fetchurl {
+    url = "https://github.com/crystal-lang/crystal/releases/download/${crystal_doc_version}/crystal-${crystal_doc_version}-docs.tar.gz";
+    hash = "sha256-OC36durhxEh/OYztjS4r9JSqpqL8f8At7+bFXQQrzhM=";
+  };
+
+  patches = [
+    ./make.patch
+    ./friendly-docs-path.patch
+    ./enable_write_permissions.patch
+  ];
+
+  shardsFile = ./rtfm-shards.nix;
+
+  nativeBuildInputs = [ wrapGAppsHook4 gobject-introspection gi-crystal ]
+    ++ lib.optionals stdenv.isDarwin [ desktopToDarwinBundle ];
+  buildInputs = [ libadwaita webkitgtk_6_0 sqlite ];
+
+  buildTargets = [ "configure" "rtfm" "docsets" ];
+
+  doCheck = false;
+
+  preBuild = ''
+    tar -xf ${crystal-docs} && mv crystal-${crystal_doc_version}-docs/ crystal-docs/
+
+    ln -s ${gtk-docs} gtk-docs
+  '';
+
+  meta = with lib; {
```

Comment:
Could you please add the `platforms` attribute with proper values? So we know on which architecture this software is available.

Reply:
Can i set `platforms = [ "i686-linux" "x86_64-linux" ];;` because i can't test on `aarch64` and `darwin`?

Reply:
Just set the platforms the software you're trying to add supports, then the CI will test on these architectures.

Reply:
I set `platforms.linux`. Theoretically, it might run on `darwin`, maybe i'll setup OpenCore later on my PC.

Reply:
Or you can ask to the Darwin maintainers team, they are quite reactive. Set it correctly and we'll ask them.

Reply:
Done. if I understand correctly how to do it
```nix
platforms = platforms.linux ++ platforms.darwin;
```

Reply:
Yes!